IRremoteESP8266.cpp

```diff
@@ -468,40 +492,38 @@ static ETSTimer timer;
 volatile irparams_t irparams;
 
 static void ICACHE_FLASH_ATTR read_timeout(void *arg) {
-    os_intr_lock();
-    if (irparams.rawlen) {
-		irparams.rcvstate = STATE_STOP;
-    }
-	os_intr_unlock();
+os_intr_lock();
+  if (irparams.rawlen) {
+    irparams.rcvstate = STATE_STOP;
+  }
+  os_intr_unlock();
 }
 
 static void ICACHE_FLASH_ATTR gpio_intr() {
-    uint32_t gpio_status = GPIO_REG_READ(GPIO_STATUS_ADDRESS);
-    GPIO_REG_WRITE(GPIO_STATUS_W1TC_ADDRESS, gpio_status);
-
-	if (irparams.rcvstate == STATE_STOP) {
-		return;
-	}	
-    static uint32_t start = 0;
-    uint32_t now = system_get_time();
-	if (irparams.rcvstate == STATE_IDLE) {
-		irparams.rcvstate = STATE_MARK;
-		irparams.rawbuf[irparams.rawlen++] = 20;		
-	}
-    else if (irparams.rawlen < RAWBUF) {
-		irparams.rawbuf[irparams.rawlen++] = (now - start) / USECPERTICK + 1;
-    }
-    start = now;
-
-    os_timer_disarm(&timer);
-    os_timer_arm(&timer, 15, 0);
+  uint32_t gpio_status = GPIO_REG_READ(GPIO_STATUS_ADDRESS);
+  GPIO_REG_WRITE(GPIO_STATUS_W1TC_ADDRESS, gpio_status);
+  if (irparams.rcvstate == STATE_STOP) {
+    return;
+  }	
+  static uint32_t start = 0;
+  uint32_t now = system_get_time();
+  if  (irparams.rcvstate == STATE_IDLE) {
+    irparams.rcvstate = STATE_MARK;	
+   // irparams.rawbuf[irparams.rawlen++] = 20 ; //???? not sounds a right thing to do  PLEASE VERIFY  !!
```

Comment:
Next time please check the commit which added this line (using 'Blame'). This was added a year ago because someone had issues without it: https://github.com/markszabo/IRremoteESP8266/commit/f863b3cb1e48b5ba2d1e374d010d36495ad15cd7